debugtools/DDR_VM/src/com/ibm/j9ddr/view/dtfj/image/J9DDRImageProcess.java

```diff
@@ -578,24 +566,64 @@ public J9DDRBaseImageThread getThread(long id) {
 		Map<Long, Object> dtfjThreadMap = getThreadMap();
 		return (J9DDRBaseImageThread)dtfjThreadMap.get(id);
 	}
-	
+
 	private int resolveGenericSignal(int num) {
-		if ((num & J9PORT_SIG_FLAG_SIGQUIT) != 0) 	return 3;
-		if ((num & J9PORT_SIG_FLAG_SIGILL) != 0) 	return 4;
-		if ((num & J9PORT_SIG_FLAG_SIGTRAP) != 0) 	return 5;
-		if ((num & J9PORT_SIG_FLAG_SIGABRT) != 0) 	return 6;
-		if ((num & J9PORT_SIG_FLAG_SIGFPE) != 0) {
-			if (num == J9PORT_SIG_FLAG_SIGFPE_DIV_BY_ZERO) 		return 35;
-			if (num == J9PORT_SIG_FLAG_SIGFPE_INT_DIV_BY_ZERO) 	return 36;
-			if (num == J9PORT_SIG_FLAG_SIGFPE_INT_OVERFLOW) 	return 37;
+
+		if (num == 0) {
+			return 0;
+		}
+
+		Platform os = process.getPlatform();
```

Comment:
This is not intended to produce OS-specific codes: the return value should be a reasonable index into the `SIGNAL_NAMES` array. For example, `OMRPORT_SIG_FLAG_SIGQUIT` should always map to `3`.

Reply:
Yes, `getSignalName` works if the core dump is from a crash and `J9DDRImageProcess` finds the generic signal number from `j9rasProcessData.gpInfo()`; however, if it's not a crash, then it falls back to getting the OS-specific signal number from the core reader, and then `getSignalName` may have the wrong mapping. For example, imagine a core is produced on Linux with `SIGBUS` but the crash is not handled by the JVM (so there's no GPINFO). Then `getSignalNumber` will return `7` on Linux (see `kill -l`) and then `getSignalName` will map that to `SIGEMT`. In addition, `getSignalNumber` is a public method and its documentation is clearly wrong in the case of a crash.

However, I agree my fix was insufficient as I'd also have to fix `getSignalName`.

Do you think I should just revert to the prior behavior (and perhaps make an update to the documentation of `getSignalNumber` that a caller should use `getSignalName` in the case of a crash), or make a fuller fix?

BTW, `com.ibm.dtfj.image.j9.ImageProcess` alludes to all of this in its `getSignalName`:

```java
> // At this point we have a signal number to convert to a name which really should be done by
> // something platform specific like the corereader.  There is a separate problem in that sometimes
> // on a GPF we get a 'generic' signal which would need some tricky mapping. 
```

Reply:
Perhaps there should be some OS-specific mapping of the value acquired via `this.process.getSignalNumber()`, but I think fixing that should be handled separately. I expect it's unlikely the information is not available from `j9rasProcessData.gpInfo()` and so doesn't represent a problem in practice. (BTW, I think it's fine on Linux.)

Reply:
Ok, makes sense, I've pushed a new commit